WebApp-OpenIDConnect-DotNet/Controllers/SessionController.cs

```diff
@@ -21,18 +22,28 @@ public SessionController(IOptions<AzureAdB2COptions> b2cOptions)
         public AzureAdB2COptions AzureAdB2COptions { get; set; }
 
         [HttpGet]
-        public IActionResult SignIn()
+        public async Task SignIn()
         {
-            return Challenge(
-                new AuthenticationProperties { RedirectUri = "/" }, OpenIdConnectDefaults.AuthenticationScheme);
+            await HttpContext.Authentication.ChallengeAsync(
+                OpenIdConnectDefaults.AuthenticationScheme, new AuthenticationProperties { RedirectUri = "/" });
         }
 
         [HttpGet]
-        public IActionResult ResetPassword()
+        public async Task ResetPassword()
         {
-            var properties = new AuthenticationProperties() { RedirectUri = "/" };
+            var properties = new AuthenticationProperties() { RedirectUri = "/"  };
             properties.Items[AzureAdB2COptions.PolicyAuthenticationProperty] = AzureAdB2COptions.ResetPasswordPolicyId;
-            return Challenge(properties, OpenIdConnectDefaults.AuthenticationScheme);
+            await HttpContext.Authentication.ChallengeAsync(
```

Comment:
I filed https://github.com/aspnet/Mvc/issues/6133 to consider adding support for specifying the ChallengeBehavior in ChallengeResult in 2.0. For this 1.1 sample this code looks fine to me.